WalletWasabi.Fluent/ViewModels/Search/SearchResult.cs

```diff
@@ -0,0 +1,6 @@
+using System.Collections.Generic;
+
+namespace WalletWasabi.Fluent.ViewModels.Search
+{
+	public sealed record SearchResult(SearchCategory Category, IEnumerable<SearchItemViewModel> Items);
```

Comment:
Iv never seen the `record` keyword before!!!

Reply:
How come there are no properties and only a Ctor?

or are they implicit Properties?

Reply:
It's new C# 9 feature. In this case instead of :
```C#
public record Person 
{ 
    public string FirstName { get; init; } 
    public string LastName { get; init; }
    public Person(string firstName, string lastName) 
      => (FirstName, LastName) = (firstName, lastName);
    public void Deconstruct(out string firstName, out string lastName) 
      => (firstName, lastName) = (FirstName, LastName);
}
```
we can write much shorter and concise form:
```C#
public record Person(string FirstName, string LastName);
```
those objects felt like nice candidate for new `record` as it's good fit for immutable data model as we use here.

As stated in [here](https://devblogs.microsoft.com/dotnet/c-9-0-on-the-record/) records are much closer to structs, but records are still reference types.

Seems like IDE have been updated, Rider even shows record type in Xaml editor properly.

I can revert to original version if necessary https://github.com/zkSNACKs/WalletWasabi/pull/4737/commits/7e3474aa88b98fd3dca6091aecb4488e8bab5f8b

Reply:
Reverted the change to `record` maybe to soon to use it.